### DISCUSSION TITLE
Undocumented parameters on API

### DISCUSSION THREAD

**Discussion by witchdr** [DependencyTrack/dependency-track#1029] [Ideas]
When I use the API to get the list of components for a project, the list is limited to 100 components. I am using:

    GET /v1/component/project/{uuid}

I notice the front-end is using:

    /api/v1/component/project/{uuid}?searchText=&sortOrder=asc&pageSize=100&pageNumber=1

However, the documentation states:
> Every API is fully documented via Swagger 2.0.

The documentation for "component" makes no mention of these parameters:

    parameters: [
    {
      name: "uuid",
      in: "path",
      required: true,
      type: "string"
    }

I don't know a great deal about Swagger but, is it possible something could be changed to make sure the API is actually fully documented?

**Comment by stevespringett** [comment]
Refer to #840. Autogenerated swagger documentation is terrible at best. There are plans to migrate to OpenAPI 3 and rip out Swagger tools in favor of manually created docs.
